.github/workflows/pull-request.yml

```
@@ -0,0 +1,212 @@
+name: Pull Request
+
+on:
+  pull_request:
+  push:
+    branches: [master]
+
+jobs:
+  lint:
+    name: Format & Clippy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          override: true
+          profile: minimal
+          components: rustfmt, clippy
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+            yew-stdweb/target
+          key: cargo-${{ runner.os }}-${{ hashFiles('**/Cargo.toml') }}
+          restore-keys: |
+            cargo-${{ runner.os }}-
+
+      - name: Run fmt
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all -- --check
+
+      - name: Run clippy
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --all-targets -- -D warnings
+
+      - name: Run clippy - yew with all features
+        run: |
+          cd yew
+          cargo clippy --all-targets --features "cbor msgpack toml yaml" -- -D warnings
+
+      - name: Run clippy - yew-stdweb with all features
+        run: |
+          cd yew-stdweb
+          cargo clippy --all-targets --features "cbor msgpack toml yaml" -- -D warnings
+
+  check_examples:
+    name: Check Examples
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          profile: minimal
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: cargo-${{ runner.os }}-${{ hashFiles('**/Cargo.toml') }}
+          restore-keys: |
+            cargo-${{ runner.os }}-
+
+      - name: Run check
+        run: |
+          cd examples
+          cargo check --all-targets
+
+  doc_tests:
+    name: Documentation Tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: wasm32-unknown-unknown
+          override: true
+          profile: minimal
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: cargo-${{ runner.os }}-${{ hashFiles('**/Cargo.toml') }}
+          restore-keys: |
+            cargo-${{ runner.os }}-
+
+      - name: Run doctest
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --workspace --doc --target wasm32-unknown-unknown
```

Looks like using the wasm target is skipping the vast majority of doc tests. I just ran without and it looks like only `yewtil/src/fetch/request.rs` fails because it tries running a real fetch request. I think we can wrap that call in a hidden cfg block. Maybe run doctests twice? Once for wasm and once non-wasm?

```suggestion
          args: --workspace --doc
```

---

Good catch, I didn't notice that these tests were being skipped... I'll try to work things out

---

I'm not sure why but only the doctests for the macro crates work when adding `--target wasm32-unknown-unknown`...

---

Ah, it seems like rustdoc can't run doctests when for any cross-compilation target whatsoever and the macro crates work because of their special nature.
I think we should mark the affected snippets as `ignore` to indicate that they aren't tested.

---

Ah, that's why! I was wondering that too. Sounds good to me, a comment describing why would be nice too 👍

---

I just remembered there's the `no_run` attribute which is exactly what we need for this. It makes sure the example compiles but doesn't run it (the example wouldn't really run anyway).